src/libcore/cell.rs

```diff
@@ -1022,12 +1023,15 @@ impl<'b> BorrowRef<'b> {
     #[inline]
     fn new(borrow: &'b Cell<BorrowFlag>) -> Option<BorrowRef<'b>> {
         let b = borrow.get();
-        if b >= MIN_WRITING {
+        if b < UNUSED {
```

Comment:
It's not clear at all that this is testing for "if nobody is writing". Could we have some `#[inline(always)]` function or a macro like `is_writing(x: BorrowFlag) -> bool` to make this code easier to understand?

Reply:
Done.